_postgresjs/index.js

```diff
@@ -0,0 +1,239 @@
+/* eslint max-len: 0 */
+
+const postgres = require('postgres')
+
+module.exports = function(options) {
+  const concurrency = options.max
+  const INSERT_PREFIX = 'insert_test__'
+  const sql = postgres({
+    fetch_types: false,
+    user: 'postgres_bench',
+    host: 'localhost',
+    database: 'postgres_bench',
+    password: 'edgedbbenchmark',
+    port: 15432,
+    ...(options || {})
+  })
+
+  const queries = {
+    get_user,
+    get_person,
+    get_movie,
+    update_movie,
+    insert_user,
+    insert_movie,
+    insert_movie_plus
+  }
+
+  const api = {
+    benchQuery: (name, id) => queries[name](id),
+    connect: () => sql`select 1`,
+    getConnection: () => api,
+    getIDs,
+    setup,
+    cleanup
+  }
+
+  return api
+
+  async function get_user(id) {
+    const [user] = await sql`
+      select
+        u.id,
+        u.name,
+        u.image,
+        json_agg(q) as latest_reviews
+      from users u, lateral (
+        select
+          r.id as review_id,
+          r.body as review_body,
+          r.rating as review_rating,
+          json_build_object(
```

Comment:
Is postgres.js capable of decoding arrays and tuples?  If so, I'd use the asyncpg queries for fairness.

Reply:
@elprans it does arrays but not record tuples. I did do a query assembling things in the client just like the go example, but it didn't do any performance difference, and in my opinion is a bit dishonest to the fact that PostgreSQL can do so much more, so if you want to honestly compare EdgeDB to PostgreSQL - choose the best from both worlds. I can change it to match the go query or asyncpg, but imo I think it would make more sense to change them to use this query?

Reply:
The issue is that JSON strips rich type information, e.g. instead of a proper `Date` object for a `timestamp` column, you get a string, etc, which is significantly less useful for application code.  This isn't entirely obvious here, because the current benchmark code just serializes things back to JSON, but it's an important aspect nonetheless.

Reply:
We can add a variation of each query that returns JSON directly, if supported.  In fact, EdgeDB explicitly supports database-side JSON encoding of results of arbitrary queries.